res/res_pjsip_outbound_registration.c

```diff
@@ -743,6 +750,26 @@ static pj_status_t registration_client_send(struct sip_outbound_registration_cli
 	/* Add Security-Verify or Security-Client headers */
 	add_security_headers(client_state, tdata);
 
+	/*
+	* Replace the User-Agent-header if a different one should be used
+	*/
+	if (!ast_strlen_zero(client_state->user_agent)) {
+		static const pj_str_t user_agent_str = { "User-Agent", 10 };
+		pjsip_generic_string_hdr *user_agent_hdr;
+		pj_str_t user_agent_val;
+		user_agent_hdr = pjsip_msg_find_hdr_by_name(tdata->msg, &user_agent_str, NULL);
+		if (user_agent_hdr) {
+			pj_list_erase(user_agent_hdr);
+		}
```

Comment:
Yeah I missed that.  You might not want to delete the existing header until you've created the new one successfully.

Reply:
I was thinking about it in a different way: Better send no UA at all than sending a "wrong"/unintended UA. That would require the error message below to be adjusted though to reflect that no UA was sent. @mbradeen @gtjoseph What do you guys think is the best way to handle this? No UA or the default UA?

Reply:
I have changed it now to use the default UA. If that is not intended after giving it a second thought I'll just change it back.

Reply:
I think default is the way.  Having no User-Agent could cause some intermediaries to block the message.